go-controller/pkg/ovn/egressip.go

```diff
@@ -2405,7 +2415,8 @@ func (e *egressIPController) deleteEgressIPStatusSetup(name string, status egres
 
 	routerName := util.GetGatewayRouterFromNode(status.Node)
 	natPred := func(nat *nbdb.NAT) bool {
-		return nat.ExternalIDs["name"] == name && nat.ExternalIP == status.EgressIP
+		return nat.ExternalIDs["name"] == name && nat.ExternalIP == status.EgressIP &&
+			nat.LogicalPort != nil && *nat.LogicalPort == types.K8sPrefix+status.Node
```

Comment:
I don't know if this is the best approach. We would depend on `status.node` correctness and we would leave any other invalid SNATs intact.
I think it might be a better option to change https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/libovsdbops/router.go#L1107 to true so we always delete all remaining SNATs for a particular egress IP.
@flavio-fernandes @jcaamano @trozet does that make sense?

Reply:
I agree with you, @kyrtapz . change bulk to true will allow for matching more than one row by the predicate.
@pperiyasamy were you being careful not to delete too many rows? In spirit of the issue you are addressing here, I would think that removing any egressip that may be stale is an okay thing to do. So change bulk to true instead is probably a better approach.

Reply:
Thanks @kyrtapz and @flavio-fernandes , I wasn't aware of this API parameter. we can just do bulk delete here.